jep/0000/README.adoc

```diff
@@ -0,0 +1,274 @@
+| 0000
+
+| Title
+| Upgrade Guava from 11.0.1 to latest
+
+| Sponsor
+| https://github.com/basil[Basil Crow]
+
+// Use the script `set-jep-status <jep-number> <status>` to update the status.
+| Status
+| Not Submitted :information_source:
+
+| Type
+| Standards
+
+| Created
+| :bulb: Date (YYYY-MM-DD) :bulb:
+
+| BDFL-Delegate
+| TBD
+
+| Jira
+| https://issues.jenkins.io/browse/JENKINS-65988[JENKINS-65988]
+
+// Uncomment when this JEP status is set to Accepted, Rejected or Withdrawn.
+//| Resolution
+//| :bulb: Link to relevant post in the jenkinsci-dev@ mailing list archives :bulb:
+
+|===
+
+== Abstract
+
+The https://guava.dev/[Guava] library in core is upgraded from https://github.com/google/guava/releases/tag/v11.0.1[11.0.1] to the latest version.
+Occurrences of Guava types in the Jenkins API are minimized.
+Plugins are prepared to be compatible with both Guava 11.0.1 and the latest version of Guava.
+Where feasible, usages of Guava in core, supporting libraries (e.g., https://github.com/stapler/stapler[Stapler]), and plugins are replaced with Java Platform equivalents.
+
+== Specification
+
+* The https://guava.dev/[Guava] library in core is upgraded from https://github.com/google/guava/releases/tag/v11.0.1[11.0.1], which was released on January 9, 2012, to the latest version.
+** The latest version of Guava and its primary dependencies are shipped in the WAR.
+** Guava annotation library dependencies are excluded from the WAR.
+* The Guava dependency is removed from https://github.com/stapler/stapler[Stapler].
+** All usages of Guava in Stapler are replaced with Java Platform equivalents.
+* Occurrences of Guava types in the Jenkins API are minimized.
+** Guava types are removed from the Jenkins API wherever possible.
+** Signatures containing Guava types that cannot be removed from the Jenkins API without breaking compatibility are deprecated, with non-deprecated replacements added where necessary.
+* Usages of Guava in core are minimized in favor of Java Platform equivalents.
+** The https://github.com/google/guava/tree/master/guava-testlib[Guava test library] is removed from the core test suite.
+* Plugins are prepared to be compatible with both Guava 11.0.1 and the latest version of Guava.
+** Usages of APIs that have been removed in the latest version of Guava are replaced with Java Platform equivalents or compatible Guava or Apache Commons APIs.
+** Plugins that use the Guava https://guava.dev/releases/snapshot-jre/api/docs/com/google/common/cache/Cache.html[`Cache`] and https://guava.dev/releases/snapshot-jre/api/docs/com/google/common/collect/MapMaker.html[`MapMaker`] APIs are migrated to https://github.com/ben-manes/caffeine[Caffeine], and Caffeine is packaged as an https://plugins.jenkins.io/caffeine-api/[API plugin] for use by other plugins.
+** Where feasible, usages of Guava in plugins are replaced with Java Platform equivalents.
+
+=== Rollout plan
+
+In mostly chronological order, though many tasks can be parallelized:
```

Comment:
I am flattered to have JEP-227 plagiarized! :grin:

Reply:
Imitation is the sincerest form